Nuget.config

```diff
@@ -1,6 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!-- Based on https://learn.microsoft.com/en-us/nuget/consume-packages/package-source-mapping#enable-by-manually-editing-nugetconfig. -->
 <configuration>
-	<packageSources>
-		<add key="NuGet" value="https://api.nuget.org/v3/index.json" />
-	</packageSources>
+    <!-- Define the package sources, `clear` ensures no additional sources are inherited from another config file. -->
+    <packageSources>
+        <clear />
+        <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
+        <add key="orchardcore-commerce" value="https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json" />
+    </packageSources>
+
+    <!-- Define mappings by adding package patterns beneath the target source. -->
+    <!-- OrchardCore.Commerce* packages will be restored from orchardcore-commerce, everything else from nuget.org. -->
+    <packageSourceMapping>
```

Comment:
Why do we need this? Release OCC packages will be on NuGet, and NuGet restore will check all sources. Also, why do we need the Cloudsmith package source here at all (since the OCC solution won't use its own packages)?

Reply:
We need the `<packageSourceMapping>` (at least the `pattern="*"` to lock in the default to NuGet), otherwise it throws like this on `dotnet restore` if you have any other source registered:
> warning NU1507: There are 3 package sources defined in your configuration. When using central package management, please map your package sources with package source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source. The following sources are defined: NuGet, orchardcore-commerce, Couldsmith

The rest is just a bit of future proofing, e.g. in case we may want to add NuGetTest later or something like that. It's also useful as reference when users ask because they aren't familiar with cloudsmith (happened in the past).

Reply:
OK then.